src/helpers/transformUniqueAssetTraitsForPresentation.ts

```diff
@@ -0,0 +1,53 @@
+import { format } from 'date-fns';
+import { UniqueAssetTrait } from '../entities/uniqueAssets';
+
+type MappedTrait = UniqueAssetTrait & {
+  originalValue: string | number | null | undefined;
+  lowercase?: boolean;
+  disableMenu?: boolean;
+};
+
+/**
+ * Mapper function that converts various NFT trait values according to their
+ * data type and other qualities. Like if it's a link it is shortening it.
+ */
+export default function transformUniqueAssetTraitsForPresentation(
+  trait: UniqueAssetTrait
+): MappedTrait {
+  const { display_type, value } = trait;
+
+  if (display_type === 'date') {
+    // the value is in seconds with milliseconds in the decimal part
+    // formatted like Jan 29th, 2022
+    const newValue =
+      typeof value === 'number' ? format(value * 1000, 'MMM do, y') : value;
+
+    return {
+      ...trait,
+      disableMenu: true,
+      originalValue: value,
+      value: newValue,
+    };
+  }
+
+  if (display_type === 'boost_percentage') {
+    return { ...trait, originalValue: value, value: `+${value}%` };
```

Comment:
```suggestion
    trait.originalValue = value;
    trait.value = `+${value}%`
```

then `else`s and `return trait` at the end

I know this looks cool and close to the FP paradigm, but I have opinions about not using this if not very necessary.

Reply:
no need to allocate a new object etc

Reply:
Makes sense it could be shorter I think I did sth you wanted see next commit